### PR TITLE
Add testing module

### PR DESCRIPTION
**Pull request by dougwettlaufer**

### CODE REVIEW

File: testing/src/test/java/io/stargate/it/CQLTest.java [hunk 1]
```diff
@@ -0,0 +1,523 @@
...
+    return null;
+  }
+
+  private static ProgrammaticDriverConfigLoaderBuilder getDriverConfigLoaderBuilder() {
+    return DriverConfigLoader.programmaticBuilder()
+        .withBoolean(DefaultDriverOption.METADATA_TOKEN_MAP_ENABLED, false)
+        .withString(
+            DefaultDriverOption.PROTOCOL_VERSION,
+            ProtocolVersion.V4
+                .name()) // FIXME: We shouldn't have to do this, but the java-driver is not
```
Comment (dougwettlaufer) [inline testing/src/test/java/io/stargate/it/CQLTest.java:515]: Well it's not just the java driver. I saw this with cqlsh so python driver is doing it too.

**Comment by mpenick** [comment]
Would it be possible to use most of the implementation from the testing module in `stargate`? Then we could just include `Cassandra40Backend` and `Dse68Backend`.

**Comment by dougwettlaufer** [comment]
> Would it be possible to use most of the implementation from the testing module in `stargate`? Then we could just include `Cassandra40Backend` and `Dse68Backend`.

What do you mean @mpenick ? As in import the tests from stargate/stargate? As far as I can tell that's not possible.

**Comment by olim7t** [comment]
You can publish a test-jar for the stargate/testing module, then you'll be able to depend on it. Then if you design some mechanism to switch the persistence backend in the test, you should be able to reuse them.

**Comment by dougwettlaufer** [comment]
@dimas-b feel free to close or take this PR over as you're working on the CCM stuff in here.

**Comment by dimas-b** [comment]
ok. I'll keep this PR in mind when reusing `stargate/testing` in this repo

**Pull request closed without merging**